libs9s/s9soptions.cpp

```diff
@@ -3302,6 +3304,40 @@ S9sOptions::s3region() const
     return retval;
 }
 
+/**
+ * \returns The argument for the --only-cloud option
+ */
+bool
+S9sOptions::onlyCloud() const
+{
+    return m_options.contains("only_cloud");
+}
+
+bool
+S9sOptions::hasCloudProviderOption() const
+{
+    return m_options.contains("cloud_provider");
+}
+
+/**
+ * \returns The argument for the --cloud-provider option 
+ */
+S9sString
+S9sOptions::cloudProvider() const
+{
+    S9sString  retval;
```

Comment:
```suggestion
        if (m_options.contains("cloud_provider")) {
        return m_options.at("cloud_provider").toString();
    }
    if(!m_userConfig.variableValue("cloud_provider").empty()) {
        return m_userConfig.variableValue("cloud_provider");
    }
    return m_systemConfig.variableValue("cloud_provider");
```
Maybe that way? It reduces complexity (less branches, return early).